botorch/optim/optimize.py

```diff
@@ -1003,8 +1038,12 @@ def optimize_acqf_mixed(
             nonlinear_inequality_constraints=nonlinear_inequality_constraints,
             post_processing_func=post_processing_func,
             batch_initial_conditions=batch_initial_conditions,
+            gen_candidates=gen_candidates,
             ic_generator=ic_generator,
             ic_gen_kwargs=ic_gen_kwargs,
+            timeout_sec=timeout_sec,
+            retry_on_optimization_warning=retry_on_optimization_warning,
+            return_best_only=True,
```

Comment:
Why True here and False earlier?

Reply:
Because, currently `return_best_only` is only implemented for `q=1` and here we are in the loop for `q>1`.